test/topology_custom/test_tablets_merge.py

```diff
@@ -334,3 +335,52 @@ async def check():
                 await manager.api.flush_keyspace(server.ip_addr, ks)
                 await manager.api.keyspace_compaction(server.ip_addr, ks)
             await check()
+
+@pytest.mark.parametrize("racks", [2, 3])
+@pytest.mark.parametrize("rf_multiplier", [1, 2])
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_tablet_merge_cross_rack_migrations(manager: ManagerClient, racks, rf_multiplier):
+    cmdline = ['--target-tablet-size-in-bytes', '30000',]
+    config = {'error_injections_at_startup': ['short_tablet_stats_refresh_interval']}
+
+    servers = []
+    rf = racks * rf_multiplier
+    for rack_id in range(0, racks):
+        rack = f'rack{rack_id+1}'
+        servers.extend(await manager.servers_add(3, config=config, cmdline=cmdline, property_file={'dc': 'mydc', 'rack': rack}))
+
+    cql = manager.get_cql()
+    ks = await create_new_test_keyspace(cql, f"WITH replication = {{'class': 'NetworkTopologyStrategy', 'replication_factor': {rf}}} AND tablets = {{'initial': 1}}")
+    await cql.run_async(f"CREATE TABLE {ks}.test (pk int PRIMARY KEY, c blob);")
+
+    await inject_error_on(manager, "disallow_cross_rack_colocation", servers)
+
+    total_keys = 400
+    keys = range(total_keys)
+    insert = cql.prepare(f"INSERT INTO {ks}.test(pk, c) VALUES(?, ?)")
+    for pk in keys:
+        value = random.randbytes(2000)
+        cql.execute(insert, [pk, value])
```

Comment:
The test would be simpler if you triggered splits and merges using table hints (min_tablet_count) rather than data size.